src/Microsoft.Identity.Web/TokenCacheProviders/TokenCacheExtensions.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Web.TokenCacheProviders;
+using Microsoft.Identity.Web.TokenCacheProviders.Distributed;
+using Microsoft.Identity.Web.TokenCacheProviders.InMemory;
+
+namespace Microsoft.Identity.Web
+{
+    /// <summary>
+    /// Extension methods to expose a simplified developer experience for
+    /// adding token caches to MSAL.NET confidential client applications
+    /// in ASP.NET, or .NET Core, or .NET FW.
+    /// </summary>
+    public static class TokenCacheExtensions
+    {
+        /// <summary>
+        /// Use a token cache and choose the serialization part by adding it to
+        /// the services collection and configuring its options.
+        /// </summary>
+        /// <returns>The confidential client application.</returns>
+        /// <param name="app">MSAL.NET cca object.</param>
+        /// <param name="initializeCaches">Action that you'll use to add a cache serialization
+        /// to the service collection passed as an argument.</param>
+        /// <returns>The application for chaining.</returns>
+        /// <example>
+        ///
+        /// The following code adds a distributed in-memory token cache.
+        ///
+        /// <code>
+        ///  app.AddTokenCaches(services =>
+        ///  {
+        ///      // In memory distributed token cache
+        ///      // In net472, requires to reference Microsoft.Extensions.Caching.Memory
+        ///      services.AddDistributedTokenCaches();
+        ///      services.AddDistributedMemoryCache();
+        ///  });
+        /// </code>
+        ///
+        /// The following code adds a token cache based on REDIS and initializes
+        /// its configuration.
+        ///
+        /// <code>
+        ///  app.AddTokenCaches(services =>
+        ///  {
+        ///       services.AddDistributedTokenCaches();
+        ///       // Redis token cache
+        ///       // Requires to reference Microsoft.Extensions.Caching.StackExchangeRedis
+        ///       services.AddStackExchangeRedisCache(options =>
+        ///       {
+        ///           options.Configuration = "localhost";
+        ///           options.InstanceName = "Redis";
+        ///       });
+        ///  });
+        /// </code>
+        ///
+        /// </example>
+        /// <remarks>Don't use this method in ASP.NET Core. Just add use the ConfigureServices method
```

Comment:
Maybe add a 'see also' to recommend AddDistributedTokenCaches?
If you are using distributed token caches, <seealso cref="AddDistributedTokenCaches"/> is recommanded, or something like that?